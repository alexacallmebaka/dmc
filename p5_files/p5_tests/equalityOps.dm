a1:int = 0;
a2:int = 1;
b:bool;
c:bool = true;

f : () int {
	a:int;
	a = 4;
}

C : class {
};

main : () void {
  b = a1 == a2;
  b = a1 == c;
  b = b != c;
  b = "true" != "true";
  b = a1 != false;
  b = a1 == C;
  b = C != f;
}
