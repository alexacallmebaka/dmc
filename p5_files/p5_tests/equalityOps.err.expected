FATAL [16,7]-[16,14]: Invalid equality operation
FATAL [19,7]-[19,18]: Invalid equality operation
FATAL [20,13]-[20,14]: Invalid equality operand
FATAL [21,7]-[21,8]: Invalid equality operand
FATAL [21,12]-[21,13]: Invalid equality operand
Type Analysis Failed
